doc/SAI-IPv6-Segment-Routing-Update.md

```diff
@@ -0,0 +1,503 @@
+     * @validonly SAI_LOCAL_SID_ENTRY_ATTR_ENDPOINT_BEHAVIOR == SAI_LOCAL_SID_ENTRY_ENDPOINT_BEHAVIOR_DT4 or SAI_LOCAL_SID_ENTRY_ATTR_ENDPOINT_BEHAVIOR == SAI_LOCAL_SID_ENTRY_ENDPOINT_BEHAVIOR_DT6 or SAI_LOCAL_SID_ENTRY_ATTR_ENDPOINT_BEHAVIOR == SAI_LOCAL_SID_ENTRY_ENDPOINT_BEHAVIOR_DT46
+     */
+    SAI_LOCAL_SID_ENTRY_ATTR_TUNNEL_ID,
+
+    /**
+     * @brief VRF for decapsulation and specific table lookup functions
+     *
+     * @type sai_object_id_t
+     * @flags CREATE_AND_SET
+     * @objects SAI_OBJECT_TYPE_VIRTUAL_ROUTER
+     * @allownull true
+     * @default SAI_NULL_OBJECT_ID
+     * @validonly SAI_LOCAL_SID_ENTRY_ATTR_ENDPOINT_BEHAVIOR == SAI_LOCAL_SID_ENTRY_ENDPOINT_BEHAVIOR_T or SAI_LOCAL_SID_ENTRY_ATTR_ENDPOINT_BEHAVIOR == SAI_LOCAL_SID_ENTRY_ENDPOINT_BEHAVIOR_DT4 or SAI_LOCAL_SID_ENTRY_ATTR_ENDPOINT_BEHAVIOR == SAI_LOCAL_SID_ENTRY_ENDPOINT_BEHAVIOR_DT6 or SAI_LOCAL_SID_ENTRY_ATTR_ENDPOINT_BEHAVIOR == SAI_LOCAL_SID_ENTRY_ENDPOINT_BEHAVIOR_DT46
+     */
+    SAI_LOCAL_SID_ENTRY_ATTR_VRF,
+
+    /**
+     * @brief Attach a counter
+     *
+     * When it is empty, then packet hits won't be counted
+     *
+     * @type sai_object_id_t
+     * @flags CREATE_AND_SET
+     * @objects SAI_OBJECT_TYPE_COUNTER
+     * @allownull true
+     * @default SAI_NULL_OBJECT_ID
+     */
+    SAI_LOCAL_SID_ENTRY_ATTR_COUNTER_ID,
+
+    /**
+     * @brief End of attributes
+     */
+    SAI_LOCAL_SID_ENTRY_ATTR_END,
+
+    /** Custom range base value */
+    SAI_LOCAL_SID_ENTRY_ATTR_CUSTOM_RANGE_START = 0x10000000,
+
+    /** End of custom range base */
+    SAI_LOCAL_SID_ENTRY_ATTR_CUSTOM_RANGE_END
+
+} sai_local_sid_entry_attr_t;
+
+/**
+ * @brief Local SID Entry
+ */
+typedef struct _sai_local_sid_entry_t
+{
+    /**
+     * @brief Switch ID
+     *
+     * @objects SAI_OBJECT_TYPE_SWITCH
+     */
+    sai_object_id_t switch_id;
+
+    /**
+     * @brief Virtual Router ID
+     *
+     * @objects SAI_OBJECT_TYPE_VIRTUAL_ROUTER
+     */
+    sai_object_id_t vr_id;
+
+    /**
+     * @brief Length of the Locator Block part of the SID
+     */
+    sai_uint8_t locator_block_len;
+
+    /**
+     * @brief Length of the Locator Node part of the SID
+     */
+    sai_uint8_t locator_node_len;
+
+    /**
+     * @brief Length of the Function part of the SID
+     */
+    sai_uint8_t function_len;
+
+    /**
+     * @brief Length of the Args part of the SID
+     */
+    sai_uint8_t args_len;
+
+    /**
+     * @brief IPv6 prefix for Local SID
+     */
+    sai_ip6_t sid;
+
+} sai_local_sid_entry_t;
+
+```
+
+5. Added apis to create/remove and set/get attributes for local_sid_entry
+
+    - add_local_sid_entry
+    - remove_local_sid_entry
+    - set_local_sid_entry_attribute
+    - get_local_sid_entry_attribute
+    - create_local_sid_entries
+    - remove_local_sid_entries
+    - set_local_sid_entries_attribute
+    - get_local_sid_entries_attribute
+
+
+## Examples ##
+- SR Headend
+    Example configuration for H.Encaps.Red behavior
```

Comment:
SID structure (a la mySID) also be allowed per SID-list (if not per SID)

Reply:
In SRv6 meeting, we agreed to postpone this for a subsequent PR.